### PR TITLE
Add introduction

### LINKED ISSUE #29: Add an introduction
Similar to the [django-uswds-forms introduction](http://django-uswds-forms.readthedocs.io/en/latest/introduction.html), we should add a brief introduction that explains the motivation behind the project and its vision.

### PR DESCRIPTION
**Pull request by toolness**
This attempts to fix #29.

### CODE REVIEW

**Comment by toolness** [comment]
@tadhg-ohiggins can you take a quick look at this introductory documentation?  Does it make sense to you?

File: docs/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
+Introduction
+============
+
+django-email-pal attempts to solve a number of issues we've
+had when iterating on projects at `18F <18f_>`_.
```
Comment (tadhg-ohiggins) [inline docs/introduction.rst]: This will work without the angle-bracket parts, as:
```
`18F`_
```

File: docs/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Simply getting started with HTML email can be daunting due to
+the lack of standards and wide variety of email clients one must
+support. It should be easy to send HTML emails that look nice,
+without having to figure out a bunch of arcane tricks to ensure
+they don't break on popular email clients.
+
+Solutions to this problem are varied. django-email-pal gives you
+the freedom to choose your own HTML email framework if you need to,
+but it also comes with a nice solution out-of-the-box: Lee Munroe's
+`Really Simple Responsive HTML Email Template <reallysimple_>`_.
```
Comment (tadhg-ohiggins) [inline docs/introduction.rst]: Similarly, you can change this to:
```
`Really Simple Responsive HTML Email Template`_
```
and then change the reference below to:
```
.. _Really Simple Responsive HTML Email Template: https://github.com/leemunroe/responsive-html-email-template
```
and it's then easier to read the inline text. The directives at the bottom aren't as clean, but I think it's better for those to be messy and the inline links to be cleaner.

**Review by tadhg-ohiggins — APPROVED** [review]
I like it! I added some minor reST comments, but think it's good to go!

**Comment by toolness** [comment]
Woot thanks!  Just applied your suggestions and will merge.

**Pull request merged**